File: binarySearch.py

def binarySearch(aList, target, start, end):

    while start <= end:
        mid = end + start / 2

        if target == aList[mid]:
            return mid

        elif target < aList[mid]:
            binarySearch(aList, target, start, mid-1)

        else:
            binarySearch(aList, target, mid+1, end)

File: binarySearch.js


function binarySearch(list, target) {
    startIndex = 0
    endIndex = list.length - 1

    while(startIndex <= endIndex) {
        midIndex = Math.floor((startIndex + endIndex) / 2);

        if(list[midIndex] === target) {
            return midIndex;
        }
        if(list[midIndex] > target) {
            startIndex = midIndex + 1;
        }
        if(list[midIndex] < target) {
            endIndex = midIndex - 1;
        }
    }
    return -1;
}